demos/jobs_for_aws/jobs_demo.c

```diff
@@ -0,0 +1,789 @@
+ * All such JSON documents will contain this key, whose value represents the unique
+ * identifier of a Job.
+ */
+#define jobsexampleQUERY_KEY_FOR_JOB_ID             jobsexampleEXECUTION_KEY  ".jobId"
+
+/**
+ * @brief The length of #jobsexampleQUERY_KEY_FOR_JOB_ID.
+ */
+#define jobsexampleQUERY_KEY_FOR_JOB_ID_LENGTH      ( sizeof( jobsexampleQUERY_KEY_FOR_JOB_ID ) - 1 )
+
+/**
+ * @brief The query key to use for searching the Jobs document ID key in message payload
+ * from AWS IoT Jobs service.
+ *
+ * Job documents received from the AWS IoT Jobs service are in JSON format.
+ * All such JSON documents will contain this key, whose value represents the unique
+ * identifier of a Job.
+ */
+#define jobsexampleQUERY_KEY_FOR_JOBS_DOC           jobsexampleEXECUTION_KEY  ".jobDocument"
+
+/**
+ * @brief The length of #jobsexampleQUERY_KEY_FOR_JOBS_DOC.
+ */
+#define jobsexampleQUERY_KEY_FOR_JOBS_DOC_LENGTH    ( sizeof( jobsexampleQUERY_KEY_FOR_JOBS_DOC ) - 1 )
+
+/**
+ * @brief The query key to use for searching the Action key in Jobs document
+ * from AWS IoT Jobs service.
+ *
+ * This demo program expects this key to be in the Job document. It is a key
+ * specific to this demo.
+ */
+#define jobsexampleQUERY_KEY_FOR_ACTION             jobsexampleQUERY_KEY_FOR_JOBS_DOC ".action"
+
+/**
+ * @brief The length of #jobsexampleQUERY_KEY_FOR_ACTION.
+ */
+#define jobsexampleQUERY_KEY_FOR_ACTION_LENGTH      ( sizeof( jobsexampleQUERY_KEY_FOR_ACTION ) - 1 )
+
+/**
+ * @brief The query key to use for searching the Message key in Jobs document
+ * from AWS IoT Jobs service.
+ *
+ * This demo program expects this key to be in the Job document if the "action"
+ * is either "publish" or "print". It represents the message that should be
+ * published or printed, respectively.
+ */
+#define jobsexampleQUERY_KEY_FOR_MESSAGE            jobsexampleQUERY_KEY_FOR_JOBS_DOC ".message"
+
+/**
+ * @brief The length of #jobsexampleQUERY_KEY_FOR_MESSAGE.
+ */
+#define jobsexampleQUERY_KEY_FOR_MESSAGE_LENGTH     ( sizeof( jobsexampleQUERY_KEY_FOR_MESSAGE ) - 1 )
+
+/**
+ * @brief The query key to use for searching the topic key in Jobs document
+ * from AWS IoT Jobs service.
+ *
+ * This demo program expects this key to be in the Job document if the "action"
+ * is "publish". It represents the MQTT topic on which the message should be
+ * published.
+ */
+#define jobsexampleQUERY_KEY_FOR_TOPIC              jobsexampleQUERY_KEY_FOR_JOBS_DOC ".topic"
+
+/**
+ * @brief The length of #jobsexampleQUERY_KEY_FOR_TOPIC.
+ */
+#define jobsexampleQUERY_KEY_FOR_TOPIC_LENGTH       ( sizeof( jobsexampleQUERY_KEY_FOR_TOPIC ) - 1 )
+
+/**
+ * @brief Utility macro to generate the PUBLISH topic string to the
+ * DescribePendingJobExecution API of AWS IoT Jobs service for requesting
+ * the next pending job information.
+ *
+ * @param[in] thingName The name of the Thing resource to query for the
+ * next pending job.
+ */
+#define START_NEXT_JOB_TOPIC( thingName ) \
```

Comment:
Ditto to the last comment.

Reply:
[Responded](https://github.com/aws/amazon-freertos/pull/2728#discussion_r520711051) in the same comment